### PR TITLE
Support TypeScript parameter properties

### PR DESCRIPTION
**Pull request by rasenplanscher**
#177

### CODE REVIEW

**Comment by rasenplanscher** [comment]
This also enables Windows-based devs to help out, like me 😉

File: package.json [hunk 1]
```diff
@@ -27,15 +29,18 @@
     "husky": "^1.3.1",
     "marked": "^0.6.1",
     "mocha": "^6.0.1",
-    "semantic-release": "^15.13.3"
+    "npm-run-all": "^4.1.5",
+    "rimraf": "^2.6.3",
+    "semantic-release": "^15.13.3",
+    "typescript": "^3.3.4000"
```
Comment (gajus) [inline package.json]: What is typescript dependency for?

Comment (rasenplanscher) [reply package.json]: TypeScript is needed by the TypeScript parse, which is needed to check TypeScript-specific syntax like the parameter properties.

Comment (brettz9) [reply package.json]: @rasenplanscher , would you mind rebasing your changes onto `master`?

**Comment by ExE-Boss** [comment]
You should change the&nbsp;PR&nbsp;description to&nbsp;“Fixes #177”.

See&nbsp;also: https://help.github.com/en/articles/closing-issues-using-keywords

**Comment by jgoz** [comment]
Is there any way I can help with this PR? This issue is blocking my team.

File: package.json [hunk 2]
```diff
@@ -27,15 +29,18 @@
...
   },
   "engines": {
     "node": ">=4"
   },
   "husky": {
     "hooks": {
-      "post-commit": "npm run create-readme && git add README.md && git commit -m 'docs: generate docs' --no-verify",
-      "pre-commit": "npm run lint && npm run test && npm run build"
+      "post-commit": "npx run-s create-readme add-assertions commit-docs",
+      "pre-commit": "npx run-s lint test build"
```
Comment (golopot) [inline package.json]: Is this necessary? It appears that shell in windows understand `&&`.

File: package.json [hunk 3]
```diff
@@ -55,10 +60,13 @@
   },
   "scripts": {
     "add-assertions": "babel-node ./src/bin/readme-assertions",
-    "build": "rm -fr ./dist && NODE_ENV=production babel ./src --out-dir ./dist --copy-files --source-maps",
-    "create-readme": "gitdown ./.README/README.md --output-file ./README.md && npm run add-assertions",
+    "build": "run-s clear-dist transpile",
+    "clear-dist": "rimraf ./dist",
```
Comment (golopot) [inline package.json]:
```suggestion
    "prebuild": "rimraf ./dist",
```

**Pull request closed without merging**